ewatercycle/observation/grdc.py

```diff
@@ -1,92 +1,100 @@
+                Data variables: streamflow  (time) float64 ... Attributes:
+                grdc_file_name:                6335020_Q_Day.Cmd id_from_grdc:
+                6335020 file_generation_date:          2019-03-27 river_name:
+                RHINE RIVER station_name:                  REES country_code:
+                DE grdc_latitude_in_arc_degree:   51.756918
+                grdc_longitude_in_arc_degree:  6.395395 grdc_catchment_area_in_km2:
+                159300.0 altitude_masl:                 8.0 dataSetContent:
+                MEAN DAILY DISCHARGE (Q) units:                         m�/s
+                time_series:                   1814-11 - 2016-12 no_of_years:
+                203 last_update:                   2018-05-24 nrMeasurements:
+                NA
     """
-    if data_home is None:
-        data_home = os.environ.get('GRDC_DATA_HOME', None)
-    if data_home is None:
-        raise ValueError('Missing data_home argument or GRDC_DATA_HOME env var to find GRDC data files')
+    if data_home:
+        data_path = Path(data_home).expanduser().resolve()
+    else:
+        if CFG["grdc_location"]:
+            data_path = Path(CFG["grdc_location"]).expanduser().resolve()
+        else:
+            raise ValueError(
+                f'Provide the grdc path using `data_home` argument '
+                f'or using `grdc_location` in ewatercycle configuration file.'
+                )
+
+    if not data_path.exists():
+        raise ValueError(f'The grdc directory {data_path} does not exist!')
 
     # Read the raw data
-    raw_file = os.path.join(data_home, station_id + "_" + parameter + "_Day.Cmd.txt")
+    raw_file = data_path / f"{station_id}_{parameter}_Day.Cmd.txt"
+    if not raw_file.exists():
+        raise ValueError(f'The grdc file {raw_file} does not exist!')
+
     # Convert the raw data to an xarray
     metadata, df = _grdc_read(
         raw_file,
-        start=datetime.strptime(start_date, "%Y-%m-%d"),
-        end=datetime.strptime(end_date, "%Y-%m-%d"))
+        start=get_time(start_time).date(),
+        end=get_time(end_time).date())
 
-    # Create the xarray dataset
-    ds = df.to_xarray()
-    ds.attrs = metadata
+    # Add start/end_time to metadata
+    metadata["UserStartTime"] = start_time
+    metadata["UserEndTime"] = end_time
 
-    return ds
+    # Add number of missing data to metadata
+    metadata["nrMissingData"] = _count_missing_data(df)
+
+    # Print info about data
+    _print_metadata(metadata)
+
+    return df, metadata
 
 
 def _grdc_read(grdc_station_path, start, end):
-    if not os.path.exists(grdc_station_path):
-        raise ValueError("Data file ", grdc_station_path, " does not exist!")
-    else:
-        with open(
-                grdc_station_path, 'r', encoding='utf-8',
-                errors='ignore') as file:
-            data = file.read()
+    with open(
+            grdc_station_path, 'r', encoding='utf-8',
+            errors='ignore') as file:
```

Comment:
To read `²` and `³` we should use `cp1252` encoding. As they match the octal 262 and 263 on the [extended ASCII codes](https://www.ascii-code.com/) table and the file when viewed with `od -c`.
```suggestion
            grdc_station_path, 'r', encoding='cp1252') as file:
```
We could also use `latin1` encoding instead, as it also works on example grdc file

Reply:
When cp1252 encoding is added here the test should also be fixed with
```patch
--- a/tests/observation/test_grdc.py
+++ b/tests/observation/test_grdc.py
@@ -26,7 +26,7 @@ def sample_grdc_file(tmp_path):
 # Country:               NA
 # Latitude (DD):       52.356154
 # Longitude (DD):      4.955153
-# Catchment area (km�):      4242.0
+# Catchment area (km²):      4242.0
 # Altitude (m ASL):        8.0
 # Next downstream station:      42424243
 # Remarks:
@@ -34,7 +34,7 @@ def sample_grdc_file(tmp_path):
 #
 # Data Set Content:      MEAN DAILY DISCHARGE (Q)
 #                        --------------------
-# Unit of measure:                  m�/s
+# Unit of measure:                  m³/s
 # Time series:           2000-01 - 2000-01
 # No. of years:          1
 # Last update:           2000-02-01
@@ -51,7 +51,7 @@ YYYY-MM-DD;hh:mm; Value
 2000-01-01;--:--;    123.000
 2000-01-02;--:--;    456.000
 2000-01-03;--:--;    -999.000'''
-    with open(fn, 'w') as f:
+    with open(fn, 'w', encoding='cp1252') as f:
         f.write(s)
     return fn
 
@@ -80,7 +80,7 @@ def test_get_grdc_data(tmp_path, sample_grdc_file):
             'river_name': 'SOME RIVER',
             'station_name': 'SOME',
             'time_series': '2000-01 - 2000-01',
-            'units': 'm�/s',
+            'units': 'm³/s',
             'UserEndTime': '2000-02-01T00:00Z',
             'UserStartTime': '20
```
So use utf exponent chars, but write with cp1252 encoding.

Reply:
The endocing= cp1252 is used both in module and test.